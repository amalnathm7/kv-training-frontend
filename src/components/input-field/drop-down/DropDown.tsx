import React from 'react';
import './DropDown.css';

export type DropDownPropsType = {
  value: string;
  onChange: (event) => void;
  label: string;
  placeholder: string;
  showError: boolean;
  options: string[];
};

const DropDown: React.FC<DropDownPropsType> = (props) => {
  const options = props.options.map((option) => (
    <option key={option} value={option}>
      {option}
    </option>
  ));

  return (
    <div className='dropdown-item'>
      <label className='dropdown-label'>{props.label}</label>
      <select
        onChange={props.onChange}
        className='dropdown-field'
        value={props.value}
        data-testid='drop-down-test'
      >
        <option hidden>{props.value}</option>
        {options}
      </select>
      {props.showError && (
        <label className='dropdown-error'>Enter {props.label.toLowerCase()}</label>
      )}
    </div>
  );
};

export default DropDown;
