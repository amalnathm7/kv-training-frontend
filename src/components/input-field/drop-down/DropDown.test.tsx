import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import DropDown, { DropDownPropsType } from './DropDown';

describe('Dropdown Props Test', () => {
  test('If dropdown rendered correctly', () => {
    const props: DropDownPropsType = {
      value: 'dropdown-value',
      onChange: () => {},
      label: 'dropdown-label',
      placeholder: 'dropdown-placeholder',
      showError: false,
      options: ['option 1', 'option 2']
    };

    const element = render(<DropDown {...props} />);

    expect(element).toMatchSnapshot();
  });

  test('If dropdown rendered correctly with error', () => {
    const props: DropDownPropsType = {
      value: 'dropdown-value',
      onChange: () => {},
      label: 'dropdown-label',
      placeholder: 'dropdown-placeholder',
      showError: true,
      options: ['option 1', 'option 2']
    };

    const element = render(<DropDown {...props} />);

    expect(element).toMatchSnapshot();
  });
});
