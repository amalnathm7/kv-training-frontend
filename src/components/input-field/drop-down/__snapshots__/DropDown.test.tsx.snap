// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown Props Test If dropdown rendered correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="dropdown-item"
      >
        <label
          class="dropdown-label"
        >
          dropdown-label
        </label>
        <select
          class="dropdown-field"
          data-testid="drop-down-test"
        >
          <option
            hidden=""
          >
            dropdown-value
          </option>
          <option
            value="option 1"
          >
            option 1
          </option>
          <option
            value="option 2"
          >
            option 2
          </option>
        </select>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="dropdown-item"
    >
      <label
        class="dropdown-label"
      >
        dropdown-label
      </label>
      <select
        class="dropdown-field"
        data-testid="drop-down-test"
      >
        <option
          hidden=""
        >
          dropdown-value
        </option>
        <option
          value="option 1"
        >
          option 1
        </option>
        <option
          value="option 2"
        >
          option 2
        </option>
      </select>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Dropdown Props Test If dropdown rendered correctly with error 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="dropdown-item"
      >
        <label
          class="dropdown-label"
        >
          dropdown-label
        </label>
        <select
          class="dropdown-field"
          data-testid="drop-down-test"
        >
          <option
            hidden=""
          >
            dropdown-value
          </option>
          <option
            value="option 1"
          >
            option 1
          </option>
          <option
            value="option 2"
          >
            option 2
          </option>
        </select>
        <label
          class="dropdown-error"
        >
          Enter 
          dropdown-label
        </label>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="dropdown-item"
    >
      <label
        class="dropdown-label"
      >
        dropdown-label
      </label>
      <select
        class="dropdown-field"
        data-testid="drop-down-test"
      >
        <option
          hidden=""
        >
          dropdown-value
        </option>
        <option
          value="option 1"
        >
          option 1
        </option>
        <option
          value="option 2"
        >
          option 2
        </option>
      </select>
      <label
        class="dropdown-error"
      >
        Enter 
        dropdown-label
      </label>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
